src/app/api/actions/service.ts

```diff
@@ -1,7 +1,25 @@
 import { prisma } from '@/db';
+import { Action } from '@prisma/client';
 
 export default class ActionService {
   static async find() {
-    return await prisma.action.findMany();
+    return await prisma.action.findMany({ orderBy: { createdAt: 'desc' } });
+  }
+
+  static async create(data: Action): Promise<Action> {
+    return prisma.action.create({
+      data,
+    });
+  }
+
+  static async update(id: string, data: Partial<Action>) {
+    return await prisma.action.update({
+      where: { id },
+      data,
+    });
+  }
+
+  static async delete(id: string) {
```

Comment:
Consider making action optional for donations and removing you want to delete action from all donations before deleting the action (foreign key constraint)